### PR TITLE
feat(connector): [Cybersource] Use connector_request_reference_id as reference to the connector

### LINKED ISSUE #2296: [FEATURE]: [Cybersource] Use `connector_request_reference_id` as reference to the connector
### :memo: Feature Description 

- We transmit our reference id from Hyperswitch to the connector, enabling merchants to subsequently locate the payment within the connector dashboard.
- Presently, there is a lack of uniformity in this process, as some implementations employ `payment_id` while others utilize `attempt_id`.
- This inconsistency arises from the fact that `payment_id` and `attempt_id` correspond to two distinct entities within our system.

### :hammer: Possible Implementation 

- To mitigate this confusion, we've introduced a new field called `connector_request_reference_id` in our RouterData. 
- This id should replace them as a reference id to the connector for payment request. 
- One might need to have exposure to api docs of the connector for which it is being implemented.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2052

:bookmark: Note : All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by sahalsaad**
Add reference with value of connector_request_reference_id in Cybersource transformers.

Fixes #2296

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/cybersource/transformers.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ pub struct CybersourcePaymentsRequest {
     processing_information: ProcessingInformation,
     payment_information: PaymentInformation,
     order_information: OrderInformationWithBill,
+    reference: String,
```
Comment (sahalsaad) [inline crates/router/src/connector/cybersource/transformers.rs:24]: I am not sure if this is correct. Feel free to request modification if needed. Thank you.

Comment (prasunna09) [reply crates/router/src/connector/cybersource/transformers.rs:24]: @sahalsaad As per api docs of [cybersource](https://developer.cybersource.com/api-reference-assets/index.html#payments) , `code` (clientReferenceInformation->code ) is merchant reference. Can you please modify accordingly?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/cybersource/transformers.rs (hunk 1 above)
Comment (sahalsaad) [reply crates/router/src/connector/cybersource/transformers.rs]: Updated the code to `ClientReferenceInformation`. Feel free to let me know if still need another modification. This is the first time I writing rust code.

**Review by prasunna09 — APPROVED** [review]

File: crates/router/src/connector/cybersource/transformers.rs [hunk 2]
```diff
@@ -278,7 +290,7 @@ pub struct CybersourcePaymentsResponse {
     client_reference_information: Option<ClientReferenceInformation>,
 }
 
-#[derive(Default, Debug, Clone, Deserialize, Eq, PartialEq)]
+#[derive(Default, Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
```
Comment (SanchithHegde) [inline crates/router/src/connector/cybersource/transformers.rs:293]: Do we absolutely need `Serialize` to be derived here, or can we remove it?

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

Thanks for the PR, @sahalsaad!

File: crates/router/src/connector/cybersource/transformers.rs (hunk 2 above)
Comment (prasunna09) [reply crates/router/src/connector/cybersource/transformers.rs:293]: We need this because, here we are reusing this struct in PaymentsRequest.

**Review by SanchithHegde — APPROVED** [review]

**Comment by VedantKhairnar** [comment]
Hey @sahalsaad  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**